components/ipfs/ipfs_service.h

```diff
@@ -81,6 +82,13 @@ class IpfsService : public KeyedService,
       base::OnceCallback<void(bool, const ipfs::NodeInfo&)>;
   using GarbageCollectionCallback =
       base::OnceCallback<void(bool, const std::string&)>;
+  // Local pins
+  using AddPinCallback =
+      base::OnceCallback<void(bool, absl::optional<AddPinResult>)>;
+  using RemovePinCallback =
+      base::OnceCallback<void(bool, absl::optional<RemovePinResult>)>;
+  using GetPinsCallback =
+      base::OnceCallback<void(bool, absl::optional<GetPinsResult>)>;
```

Comment:
For these three callbacks let's remove bool argument and consider empty optional as failure